PixelHunter1995/WalkingAreaLib/PolygonPartition.cs

```diff
@@ -76,5 +87,46 @@ public bool Contains(Vector2 position)
         {
             return Polygons.Any(polygon => polygon.Contains(position));
         }
+
+        public bool ContainsLine(Vector2 pointA, Vector2 pointB)
+        {
+            if (!Contains(pointA) || !Contains(pointB))
+            {
+                return false;
+            }
+
+            List<Vector2> intersections = OriginalPolygon.EdgeIntersections(pointA, pointB);
+            foreach (Vector2 intersectionA in intersections)
+            {
+                foreach (Vector2 intersectionB in intersections)
```

Comment:
Shouldn't this loop just cover the intersections that come after intersectionA? Otherwise we will do duplicate work, right?